src/anonymize_proxy.js

```diff
@@ -15,8 +13,8 @@ const anonymizedProxyUrlToServer = {};
  * anonymous proxy URL or the original URL if it was already anonymous.
  */
 const anonymizeProxy = (proxyUrl, callback) => {
-    const parsedProxyUrl = parseUrl(proxyUrl);
-    if (!parsedProxyUrl.host || !parsedProxyUrl.port) {
+    const parsedProxyUrl = new URL(proxyUrl);
+    if (!parsedProxyUrl.hostname || !parsedProxyUrl.port) {
```

Comment:
When we change the parsing logic, I'm pretty sure that it will no longer work for some users so we need to keep in mind that we should only update `proxy-chain` with a major update of `browser-pool` and `apify-js`. cc @B4nan

Reply:
> When we change the parsing logic, I'm pretty sure that it will no longer work

Why wouldn't it work? This check is removed now, the protocol check is sufficient.

Reply:
I'm talking about the switch from `parseUrl` to `new URL` without all the additional props and defaults `parseUrl` added. I'm pretty sure that given the number of use cases we have, there will be some people with some edge-case proxy URLs that will start failing for them.

Reply:
> there will be some people with some edge-case proxy URLs that will start failing for them.

Then they will need to switch to WHATWG URL (which is a good change). For `proxy-chain` there's no need to have this. This rewrite will be a major anyway.

Reply:
Yeah, they will. But I want to make sure we don't forget about this and do not merge `proxy-chain` v2 into SDK v2. We can only add it to v3 because there's a lot of under the hood changes.

Thinking about it more. I'm not quite sure how this will work for all the proxy URLs that are used from the Apify Console UI. Those people won't easily change to a WHATWG URL and there might be hell on support because of this.

Reply:
> I'm not quite sure how this will work for all the proxy URLs that are used from the Apify Console UI.

I don't understand. What would change? Their actors will work exactly the same without any changes. The old `parseUrl` wraps WHATWG URL already. They can't tinker with server-side code, can they?

If you mean providing their own proxy URLs, the behavior remains unchanged.

Reply:
I guess we'll have to see :)